src/librustdoc/html/render.rs

```diff
@@ -3606,15 +3640,15 @@ fn sidebar_trait(fmt: &mut fmt::Formatter, it: &clean::Item,
 
     sidebar.push_str("<li><a href=\"#implementors\">Implementors</a></li>");
```

Comment:
Move the `sidebar_assoc_items` call below these items and i'll call it good. As-is, this will print these links *below* the assoc items listing, thus putting the "table of contents" links *between* the methods listings and the module-siblings listings, effectively hiding them.